alert-database/src/test/java/com/synopsys/integration/alert/database/api/DefaultDescriptorAccessorTest.java

```diff
@@ -0,0 +1,289 @@
+
+        RegisteredDescriptorEntity registeredDescriptorEntity = new RegisteredDescriptorEntity(name, typeId);
+        registeredDescriptorEntity.setId(1L);
+        DescriptorTypeEntity descriptorTypeEntity = new DescriptorTypeEntity(descriptorType.name());
+        descriptorTypeEntity.setId(1L);
+
+        Mockito.when(descriptorTypeRepository.findFirstByType(descriptorType.name())).thenReturn(Optional.of(descriptorTypeEntity));
+        Mockito.when(registeredDescriptorRepository.findByTypeId(Mockito.any())).thenReturn(List.of(registeredDescriptorEntity));
+        Mockito.when(descriptorTypeRepository.findById(registeredDescriptorEntity.getTypeId())).thenReturn(Optional.of(descriptorTypeEntity));
+
+        DefaultDescriptorAccessor descriptorAccessor = new DefaultDescriptorAccessor(registeredDescriptorRepository, null, null, descriptorTypeRepository);
+        List<RegisteredDescriptorModel> registeredDescriptorModelList = descriptorAccessor.getRegisteredDescriptorsByType(descriptorType);
+
+        assertEquals(1, registeredDescriptorModelList.size());
+        assertEquals(name, registeredDescriptorModelList.get(0).getName());
+        assertEquals(descriptorType, registeredDescriptorModelList.get(0).getType());
+    }
+
+    @Test
+    public void getRegisteredDescriptorsByTypeMissingDescriptorTypeTest() throws Exception {
+        final String name = "name-test";
+        final Long typeId = 1L;
+        final DescriptorType descriptorType = DescriptorType.CHANNEL;
+
+        RegisteredDescriptorRepository registeredDescriptorRepository = Mockito.mock(RegisteredDescriptorRepository.class);
+        DescriptorTypeRepository descriptorTypeRepository = Mockito.mock(DescriptorTypeRepository.class);
+
+        RegisteredDescriptorEntity registeredDescriptorEntity = new RegisteredDescriptorEntity(name, typeId);
+        registeredDescriptorEntity.setId(1L);
+        DescriptorTypeEntity descriptorTypeEntity = new DescriptorTypeEntity(descriptorType.name());
+        descriptorTypeEntity.setId(1L);
+
+        Mockito.when(descriptorTypeRepository.findFirstByType(descriptorType.name())).thenReturn(Optional.empty());
+        Mockito.when(descriptorTypeRepository.save(Mockito.any())).thenReturn(descriptorTypeEntity);
+        Mockito.when(registeredDescriptorRepository.findByTypeId(Mockito.any())).thenReturn(List.of(registeredDescriptorEntity));
+        Mockito.when(descriptorTypeRepository.findById(registeredDescriptorEntity.getTypeId())).thenReturn(Optional.of(descriptorTypeEntity));
+
+        DefaultDescriptorAccessor descriptorAccessor = new DefaultDescriptorAccessor(registeredDescriptorRepository, null, null, descriptorTypeRepository);
+        List<RegisteredDescriptorModel> registeredDescriptorModelList = descriptorAccessor.getRegisteredDescriptorsByType(descriptorType);
+
+        assertEquals(1, registeredDescriptorModelList.size());
```

Comment:
How is this any different than the previous test? It looks like the assertions do not match the test name

Reply:
This test is testing for an Optional.empty case that is used by one of the private methods to increase its coverage, it is being passed on line 141 and as a result needed some other mocks. Because it will create a new descriptorTypeEntity I used the same setup as in the previous test after that.